iModelCore/ECDb/Tests/NonPublished/InstanceReaderTest.cpp

```diff
@@ -0,0 +1,270 @@
+    ASSERT_STRCASEEQ(stmt.GetValueText(i++), "{\"X\":4.0,\"Y\":5.0,\"Z\":6.0}"); // return as json
+    ASSERT_EQ(memcmp(stmt.GetValueBlob(i++), &kBi[0], kBiLen), 0);
+    ASSERT_EQ(stmt.GetValueInt64(i++), kL);
+    ASSERT_TRUE(stmt.GetValueDateTime (i++).Equals(kDt, true));
+    ASSERT_EQ(stmt.GetValueBoolean(i++), kB);
+
+}
+
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//+---------------+---------------+---------------+---------------+---------------+------
+TEST_F(InstanceReaderFixture, prop_exists) {
+    ASSERT_EQ(SUCCESS, SetupECDb("PROP_EXISTS.ecdb", SchemaItem(
+        R"xml(<ECSchema schemaName='TestSchema' alias='ts' version='1.0' xmlns='http://www.bentley.com/schemas/Bentley.ECXML.3.1'>
+                   <ECSchemaReference name="ECDbMap" version="02.00" alias="ecdbmap" />
+                   <ECEntityClass typeName="Base" modifier="Abstract">
+                        <ECCustomAttributes>
+                            <ClassMap xmlns="ECDbMap.02.00">
+                                <MapStrategy>TablePerHierarchy</MapStrategy>
+                            </ClassMap>
+                            <JoinedTablePerDirectSubclass xmlns="ECDbMap.02.00"/>
+                        </ECCustomAttributes>
+                        <ECProperty propertyName="Prop1" typeName="string" />
+                        <ECProperty propertyName="Prop2" typeName="int" />
+                    </ECEntityClass>
+                    <ECEntityClass typeName="Sub">
+                      <BaseClass>Base</BaseClass>
+                      <ECProperty propertyName="SubProp1" typeName="string" />
+                      <ECProperty propertyName="SubProp2" typeName="int" />
+                     </ECEntityClass>
+               </ECSchema>)xml")));
+    m_ecdb.SaveChanges();
+
+    if ("case sensitive check") {
+        const auto sql =
+            "SELECT "
+            "   PROP_EXISTS(ec_classid('ts.Base'), 'ECClassId'   ),"
+            "   PROP_EXISTS(ec_classid('ts.Base'), 'ECInstanceId')," 
+            "   PROP_EXISTS(ec_classid('ts.Base'), 'Prop1'       ),"
+            "   PROP_EXISTS(ec_classId('ts.base'), 'Prop2'       )," 
+            "   PROP_EXISTS(ec_classid('ts.Base'), 'SubProp1'    ),"
+            "   PROP_EXISTS(ec_classid('ts.Base'), 'SubProp2'    ),"
+            "   PROP_EXISTS(ec_classid('ts.Sub'),  'ECClassId'   )," 
+            "   PROP_EXISTS(ec_classid('ts.Sub'),  'ECInstanceId'),"
+            "   PROP_EXISTS(ec_classid('ts.Sub'),  'Prop1'       ),"
+            "   PROP_EXISTS(ec_classId('ts.Sub'),  'Prop2'       ),"
+            "   PROP_EXISTS(ec_classid('ts.Sub'),  'SubProp1'    ),"
+            "   PROP_EXISTS(ec_classid('ts.Sub'),  'SubProp2'    )";
+
+        ECSqlStatement stmt;   
+        ASSERT_EQ(ECSqlStatus::Success, stmt.Prepare(m_ecdb, sql));
+
+        ASSERT_EQ(stmt.Step(), BE_SQLITE_ROW);
+
+        int i = 0;
+        ASSERT_EQ(stmt.GetValueInt(i++), 1);
```

Comment:
Could we use boolean instead of int for this? Or is it intentional?

Reply:
It is the return value of SQLite function which is integer as SQLite has no boolean data type. So, my test captured return value of the function. Though the GetValueBoolean() will work as well.

Reply:
i have added the test.